### PR TITLE
Eo3 test fixtures

### PR DESCRIPTION
**Pull request by SpacemanPaul**
### Reason for this pull request

Test fixtures were mostly EO-based and thus could not be used with the postgis driver (or were telemetry dataset based and may break on postgis driver in future).

### Proposed changes

- Add EO3 test fixtures
- Make EO3 versions of EO-telemetry-dataset-fixture-based tests
- Fix/cleanup various SQLAlchemy bugs and issues from recent postgis driver refactor

 - [x] Tests added / passed
 - [x] Fully documented, including `docs/about/whats_new.rst` for all changes

### CODE REVIEW

File: datacube/index/fields.py [hunk 1]
```diff
@@ -29,6 +29,9 @@ class OrExpression(Expression):
     def __init__(self, *exprs):
         super(OrExpression, self).__init__()
         self.exprs = exprs
+        for expr in self.exprs:
+            self.field = expr.field
+            break
```
Comment (omad) [inline datacube/index/fields.py]: Maybe I'm slow, but this took me a while to parse what it was doing. A for loop, but just to extract the first expression's field to set as the Expression field.

It seems a bit weird. Why does an Or expression even need a field? Surely it's just made up of other expressions, not fields...

If it's just to make a test pass, maybe the test needs fixing.

File: integration_tests/index/search_utils.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import csv
```
Comment (omad) [inline integration_tests/index/search_utils.py:5]: I think we're supposed to have our copyright message at the top of every file. Please and thank you.

When I'm next bored I might add it as another pre-commit hook [ODC issue referencehttps://github.com/opendatacube/datacube-core/issues/1310).

**Review by omad — CHANGES_REQUESTED** [review]
Two (hopefully) minor things. Otherwise looks good, it's nice to have more current tests. thank you Paul

File: datacube/index/fields.py (hunk 1 above)
Comment (SpacemanPaul) [reply datacube/index/fields.py]: What I need to be able to do is determine which tables are involved in the query so I can string together a valid from/join clause.  I can get the table from a field.  A simple expression always has a field, and an OR expression has multiple simple expressions, which in practice always share the same field:  In ODC, OR expressions are built from load queries like: `dc.load(search_field=['value1', 'value2'])` resulting in SQL like `(search_field='value1' or search_field='value2')`.

I'm not aware of a way to get an ODC OR expression combining multiple fields (e.g. `(search_field='value' or other_search_field='value2')`) so this code should work - but certainly could be written more clearly/elegantly.  Hell, even a comment would help.

The whole field/expression and query building infrastructure needs a good re-work.  The repurposing of the postgres driver infrastructure for the declarative classes is still pretty hacky.

File: integration_tests/index/search_utils.py (hunk 2 above)
Comment (SpacemanPaul) [reply integration_tests/index/search_utils.py:5]: Yes indeed, good catch.

**Review by omad — APPROVED** [review]
Thanks Paul!

The `OrExpressions` thing does in my head a bit, but, the comment makes that bit of code at least seem less crazy.

**Pull request merged**